woocommerce/woocommerce-ios
# Product creation AI: Reuse detected language on retries

Closes: #11653 


## Description

This PR adds an improvement to only request language detection for product creation AI on first try. This will improve the response for subsequent requests.

Also: updated unit tests to make them shorter.

## Testing instructions

It's not straightforward to replicate the case where the product generation fails. We can update the method `ProductDetailPreviewViewModel` to force the error state like so:

<img width="627" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/e6141117-6a6e-4f79-a732-a10a856d7262">

- Build the app and log in to a store eligible for Jetpack AI.
- Navigate to the Products tab, select "+" then "Create a product with AI".
- Follow the steps to generate a product.
- After reaching the Preview screen, an alert should be displayed saying there was an error generating product details. Confirm in Proxyman that there was a request to identify the language in the title and keywords immediately before that.
- Tap Retry in the alert. Confirm in Proxyman that no other request to identify the language is sent. 
- Tap Cancel in the alert. Try generate the product again.
- Confirm in Proxyman that a request to identify the language is sent again.

## Screenshots

N/A

---
